src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_arm_utils.py

```diff
@@ -59,10 +62,21 @@ def arm_deploy_template(resource_group_name, registry_name, location, storage_ac
     parameters = _parameters(registry_name, location, storage_account_name)
     properties = DeploymentProperties(template=template, parameters=parameters, mode='incremental')
 
+    return _arm_deploy_template(resource_group_name, properties)
+
+def _arm_deploy_template(resource_group_name, properties, index=0):
```

Comment:
Fixed. It is basically a validate and retry function to avoid deployment name conflict. Index is a number added to deployment names.